File: youlexuan_cart_interface/src/main/java/com/offcn/cart/service/CartService.java
package com.offcn.cart.service;

import com.offcn.entity.Cart;

import java.util.List;

/**
 * @author zhangjian
 * @email 13120082225@163.com
 * @date 2020/4/24
 */
public interface CartService {
    /**
     * 添加商品到购物车
     * @param cartList 原购物车商品集合
     * @param itemId  sku编号
     * @param num    购买数量
     * @return
     */
    List<Cart> addGoodsToCartList(List<Cart> cartList, Long itemId, Integer num );

    /**
     * 处理购物车数据一致性
     * @param carts
     * @return
     */
    List<Cart> checkCartListDataInfos(List<Cart> carts);
}

File: youlexuan_cart_web/src/main/webapp/js/service/cartService.js
//购物车服务层
app.service('cartService', function ($http) {
    //购物车列表
    this.findCartList = function () {
        alert("33");
        return $http.get('cart/findCartList.do');
    };
    //添加商品到购物车
    this.addGoodsToCartList=function(itemId,num){
        return $http.get('cart/addGoodsToCartList.do?itemId='+itemId+'&num='+num);
    };

    //计算购物车页面中，所有商品的价格总和
    this.sum = function (cartList) {
        //合计实体
        var totalValue = {totalNum: 0, totalMoney: 0.00};
        for (var i = 0; i < cartList.length; i++) {
            var cart = cartList[i];
            for (var j = 0; j < cart.orderItemList.length; j++) {
                //购物车明细
                var orderItem = cart.orderItemList[j];
                totalValue.totalNum += orderItem.num;
                totalValue.totalMoney += orderItem.totalFee;
            }
        }
        return totalValue;
    }
});
